code/src/containers/tables/orders-table/components/orders-table-head/OrdersTableHead.test.tsx

```diff
@@ -86,16 +70,97 @@ describe("OrderTableHead", () => {
     });
   });
 
-  test("sort icon has correct initial and updated classes", () => {
-    renderComponent("ordersTable.columns.totalPrice");
-    const sortButton = screen.getByTestId("ArrowDownwardIcon");
+  test("updates searchParams on sort", async () => {
+    renderComponent(sortableFields);
+    const sortButtonCreatedAt = screen.getAllByTestId("ArrowDownwardIcon")[0];
+    const sortButtonTotalPrice = screen.getAllByTestId("ArrowDownwardIcon")[1];
+
+    fireEvent.click(sortButtonCreatedAt);
+
+    let params = new URLSearchParams(mockSetSearchParams.mock.calls[0][0]);
+    expect(params.get("sort")).toBe("createdAt,asc");
+
+    fireEvent.click(sortButtonCreatedAt);
+    await waitFor(() => {
+      params = new URLSearchParams(mockSetSearchParams.mock.calls[1][0]);
+      expect(params.get("sort")).toBe("createdAt,desc");
+    });
+
+    fireEvent.click(sortButtonTotalPrice);
+    await waitFor(() => {
+      params = new URLSearchParams(mockSetSearchParams.mock.calls[2][0]);
+      expect(params.get("sort")).toBe("total,asc");
+    });
+
+    fireEvent.click(sortButtonCreatedAt);
+    await waitFor(() => {
+      params = new URLSearchParams(mockSetSearchParams.mock.calls[3][0]);
+      expect(params.get("sort")).toBe("createdAt,asc");
+    });
```

Comment:
Try to reuse this logic. Maybe create function that will perform step by index

Reply:
Created helper func to avoid code repeating